book/src/generalized-ownership.md

```diff
@@ -0,0 +1,284 @@
+
+Using the account's pubkey identity as the asset owner identity means that all
+the asset transfer APIs must be re-implemented by the program code.
+
+The proposed solution is to use a pointer to an owner identity.
+
+```rust,ignore
+// This state is owned by the Owner program
+enum OwnerState {
+    // the owner identity must provide a valid signature for this pubkey
+    Unique {
+        // The actual owner key.
+        owner: Pubkey, 
+        // The item program instance.
+        item: Pubkey
+    },
+    //...
+}
+```
+
+* OwnerState::Unique.owner - The actual OwnerState’s account’s pubkey is not used to
+distinguish ownership.  Only the `OwnerState::Unique.owner` pubkey identifies the
+ownership identity.
+
+```rust,ignore
+// This state is owned by the MyUniqueTokenState
+struct MyUniqueTokenState {
+    // This points to an OwnerState structure that identifies the actual owner
+    owner: Pubkey,
+}
+```
+
+* MyUniqueStokenState.owner - address of the OwnerState that identifies the
+current owner.
+
+### OwnerState::Transfer
+
+* account[0] - RW - The OwnerState.
+* account[1] - R - Current owner Pubkey, needs a signature. Doesn’t need an
+allocated Account.
+* data - New owner Pubkey. Doesn’t need an allocated Account.
+
+### MyTokenState::Init
+
+* account[0] - RW - MyUniqueTokenState. MyUniqueTokenState initializes the
+owner once with `account[1]`.  This program must never modify that field in any
+future state transitions.
+* account[1] - R - The OwnerState.  OwnerState::Unique.item must be equal to
+account[0].
+
+### OwnerState::Countable
+
+The item that is owned by this OwnerState must have a valid count `0.`
+representation that the user can create.  Units are then transferred from one
+`OwnerState::Countable` into another, and the total sum is preserved between
+transfers.
+
+```rust,ignore
+// This state is owned by the Owner program
+enum OwnerState {
+    //...
+    Countable {
+        // The owner identity must provide a valid signature for this pubkey
+        owner: Pubkey,
+        // The number of units this OwnerState represents
+        units: u64,
+        // The item program instance.
+        item: Pubkey,
+    },
+    //...
+}
+```
+
+When `OwnerState::Countable::units == 1`, this object is exactly the same as
+`OwnerState::Unique`.  The reason for a separate `OwnerState::Unique` is to
+programmatically ensure that only a single instance of this item exists.
+
+```rust,ignore
+// This state is owned by the MyTokenState
+struct MyTokenState {
+    // This points to an OwnerState structure that identifies the actual owner
+    owner: Pubkey,
+}
+
+```
+
+The MyToken program must expect many `OwnerStates` to point to the same instance
+of `MyTokenState`
+
+## Programmatic Transfer of Ownership
+
+Currently, programs have no way to trigger a transfer of ownership. A game of
+TicTacToe has no way to assign a prize to the winner.
+
+```rust,ignore
+// This state is owned by the Owner program
+enum OwnerState {
```

Comment:
Another definition of OwnerState? Are these progressively more advanced definitions? Perhaps this was all the same OwnerState definition and each field was introduced as a full OwnerState definition?